From: Rolf Eike Beer <eb@emlix.com>
Subject: mm: pagewalk: don't check vma in walk_page_range_novma()
Date: Mon, 22 Aug 2022 15:00:47 +0200

Directly call walk_pgd_range() as that is everything that will actually
happen in __walk_page_range() besides checking if the vma is set.

Link: https://lkml.kernel.org/r/21585888.EfDdHjke4D@devpool047
Signed-off-by: Rolf Eike Beer <eb@emlix.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/pagewalk.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/pagewalk.c~mm-pagewalk-dont-check-vma-in-walk_page_range_novma
+++ a/mm/pagewalk.c
@@ -517,7 +517,7 @@ int walk_page_range_novma(struct mm_stru
 
 	mmap_assert_locked(walk.mm);
 
-	return __walk_page_range(start, end, &walk);
+	return walk_pgd_range(start, end, &walk);
 }
 
 int walk_page_vma(struct vm_area_struct *vma, const struct mm_walk_ops *ops,
_
